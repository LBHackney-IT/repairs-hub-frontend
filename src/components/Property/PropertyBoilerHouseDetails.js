import PropTypes from 'prop-types'
import { usePropertyBoilerHouse } from '../../hooks/usePropertyBoilerHouse'
import Spinner from '../Spinner'
import ErrorMessage from '../Errors/ErrorMessage'

const PropertyBoilerHouseDetails = ({ boilerHouseId }) => {
  const { loading, boilerHouse, boilerHouseError } = usePropertyBoilerHouse(
    boilerHouseId
  )

  if (loading) return <Spinner resource="propertyBoilerHouseDetails" />

  if (boilerHouseError) return <ErrorMessage label={boilerHouseError} />

  return (
    <div>
      {boilerHouse !== null && (
        <div
          style={{
            background: '#00664f',
            color: 'white',
            padding: '4px 8px',
            margin: '15px 0',
            fontSize: '1rem',
            display: 'inline-block',
          }}
        >
          BoilerHouse:{' '}
          <a
            className="govuk-link"
            style={{
              color: 'white',
            }}
            href={`/properties/${boilerHouse.propertyReference}`}
          >
            {boilerHouse.addressLine1}
          </a>
        </div>
      )}
    </div>
  )
}

PropertyBoilerHouseDetails.propTypes = {
  boilerHouseId: PropTypes.string.isRequired,
}

export default PropertyBoilerHouseDetails
