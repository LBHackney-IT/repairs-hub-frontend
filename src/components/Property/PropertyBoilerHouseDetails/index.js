import PropTypes from 'prop-types'
import { usePropertyBoilerHouse } from '../../../hooks/usePropertyBoilerHouse'
import Spinner from '../../Spinner'
import ErrorMessage from '../../Errors/ErrorMessage'

const renderLegalDisrepair = (isInLegalDisrepair) => {
  return (
    isInLegalDisrepair && (
      <WarningInfoBox
        header="This property is currently under legal disrepair"
        text="Before raising a work order you must call the Legal Disrepair Team"
      />
    )
  )
}

const PropertyBoilerHouseDetails = ({ boilerHouseId }) => {
  const { loading, boilerHouse, boilerHouseError } = usePropertyBoilerHouse(
    boilerHouseId
  )

  {loading ? <Spinner resource="propertyBoilerHouseDetails"/> : renderLegalDisrepair(isInLegalDisrepair)}

  if (boilerHouseError) return <ErrorMessage label={boilerHouseError} />

  return (
    <li className="bg-dark-green" data-testid="boiler-house-details">
      BoilerHouse:{' '}
      <a
        data-testid="boiler-house-details-link"
        className="govuk-link"
        style={{
          color: 'white',
        }}
        href={`/properties/${boilerHouse.propertyReference}`}
      >
        {boilerHouse.addressLine1}
      </a>
    </li>
  )
}

PropertyBoilerHouseDetails.propTypes = {
  boilerHouseId: PropTypes.string.isRequired,
}

export default PropertyBoilerHouseDetails
