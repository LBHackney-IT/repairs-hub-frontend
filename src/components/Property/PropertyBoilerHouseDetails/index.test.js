import axios from 'axios'

import PropertyBoilerHouseDetails from '.'
import { render, screen, waitFor } from '@testing-library/react'

jest.mock('axios', () => jest.fn())

describe('PropertyBoilerHouseDetails', () => {
  it('it shows an error message when request fails', async () => {
    axios.mockImplementationOnce(() =>
      Promise.reject({
        response: {
          status: 404,
          data: {
            message: 'error message',
          },
        },
      })
    )

    const boilerHouseId = '1234'

    const { asFragment } = render(
      <PropertyBoilerHouseDetails boilerHouseId={boilerHouseId} />
    )

    await waitFor(async () => {
      expect(screen.getByTestId(/error-message/i)).toHaveTextContent(
        'Error loading boiler house details status 404 with message: error message'
      )
    })

    expect(asFragment()).toMatchSnapshot()
  })

  it('it shows boiler house details', async () => {
    const mockResponse = {
      propertyReference: '00023404',
      address: {
        shortAddress: '16 Pitcairn House St Thomass Square',
        postalCode: 'E9 6PT',
        addressLine: '16 Pitcairn House St Thomass Square',
        streetSuffix: null,
      },
      hierarchyType: {
        levelCode: null,
        subTypeCode: null,
        subTypeDescription: 'Dwelling',
      },
      tmoName: 'London Borough of Hackney',
      canRaiseRepair: false,
    }

    axios.mockResolvedValue({
      data: mockResponse,
    })

    const boilerHouseId = '1234'

    const { asFragment } = render(
      <PropertyBoilerHouseDetails boilerHouseId={boilerHouseId} />
    )

    await waitFor(async () => {
      expect(screen.getByTestId('boiler-house-details-link')).toHaveTextContent(
        mockResponse.address.addressLine
      )
      expect(screen.getByTestId('boiler-house-details-link')).toHaveAttribute(
        'href',
        `/properties/${mockResponse.propertyReference}`
      )
    })

    expect(asFragment()).toMatchSnapshot()
  })
})
