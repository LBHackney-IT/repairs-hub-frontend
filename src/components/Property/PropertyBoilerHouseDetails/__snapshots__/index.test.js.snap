// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyBoilerHouseDetails shows an error message when request fails 1`] = `
<DocumentFragment>
  <span
    class="govuk-error-message lbh-error-message"
    data-error-id=""
    data-testid="error-message"
  >
    <span
      class="govuk-visually-hidden"
    >
      Error:
    </span>
     Error loading boiler house details status 404 with message: error message
  </span>
</DocumentFragment>
`;

exports[`PropertyBoilerHouseDetails shows boiler house details 1`] = `
<DocumentFragment>
  <li
    class="bg-dark-green"
    data-testid="boiler-house-details"
  >
    BoilerHouse: 
    <a
      class="govuk-link"
      data-testid="boiler-house-details-link"
      href="/properties/00023404"
      style="color: white;"
    >
      16 Pitcairn House St Thomass Square
    </a>
  </li>
</DocumentFragment>
`;
